scipy/optimize/tests/test__numdiff.py

```diff
@@ -185,6 +185,8 @@ def fun_non_numpy(self, x):
         return math.exp(x)
 
     def jac_non_numpy(self, x):
+        # Input can be float x or numpy.array([x]). Cast it all into numpy.array(x).
```

Comment:
Again, this feels like a bug in the caller - it really ought to be certain what shape it calls this with.

Reply:
This is from x0 in scipy/optimize/_numdiff.py. The doc says
```
    x0 : array_like of shape (n,) or float
        Point at which to estimate the derivatives. Float will be converted
        to a 1-d array.
```q
The `Float will be converted to a 1-d array.` is weird.

Reply:
Honestly, it feels like what should actually happen is something like:
```
x0_flat = x0.ravel()

x0_revised = do_linalgy_stuff_that_needs_x_1d(x0_flat)

f(x0_revised.reshape(x0.shape))
```

Reply:
I'll perhaps try a patch for that in the next few weeks

Reply:
The purpose behind that docstring is to let the user know that `f` is going to be passed an array regardless of whether `x` is an array or a float.